### PR TITLE
Stefan/ncc add

### PR DESCRIPTION
**Pull request by ste00martin**
this is an interactive PR with some questions for the authors:
1. should we turn off semicolons?
2. should the `compiled` folder be checked into GitHub?

### CODE REVIEW

**Comment by ste00martin** [comment]
One more ask from @punov - can we enable git merge+squash and disable the vanilla git merge for a cleaner main branch history?

**Comment by hmn8te** [comment]
I'm okay with merge + squash. I tend to write cute commits myself, but I'm okay either way.

**Comment by hmn8te** [comment]
I don't see any redflags here @ste00martin  Looks good to merge. I would tend to make a PR for each of these changes, but since they are all small and seem to be working I'm happy to merge.

**Review by hmn8te — APPROVED** [review]
LGTM

File: .eslintrc.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
         "@typescript-eslint"
     ],
     "rules": {
+    //   "semi": ["error", "never"],
```
Comment (hmn8te) [inline .eslintrc.json:19]: Do we need this?

**Review by hmn8te — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (ste00martin) [reply .eslintrc.json:19]: This was more of a proposal for us to start not using semicolons; ill remove it

**Pull request merged**